lib/private/shell.ts

```diff
@@ -29,39 +29,42 @@ export async function shell(command: string[], options: ShellOptions = {}): Prom
       child.stdin!.end();
     }
 
-    const stdout = new Array<any>();
-    const stderr = new Array<any>();
+    const stdoutChunks = new Array<any>();
+    const stderrChunks = new Array<any>();
 
-    // Both write to stdout and collect
     child.stdout!.on('data', (chunk) => {
       if (!options.quiet) {
-        process.stdout.write(chunk);
+        globalOutputHandler.publishEvent(chunk, EventType.DEBUG);
       }
-      stdout.push(chunk);
+      stdoutChunks.push(chunk);
```

Comment:
We *probably* want to stream output to the logger. Docker builds can take 10 minutes, and it's not very helpful if you're sitting there, waiting, not knowing what's going on...

And then after 10 minutes you get a 2MB log dump of everything that happepend.